test/e2e/clients/python/framework/context.py

```diff
@@ -0,0 +1,310 @@
+                        core_v1.patch_namespaced_service(
+                            name=manifest_name, namespace=namespace, body=manifest
+                        )
+                    else:
+                        raise
+        elif api_version == "apps/v1":
+            apps_v1 = self.get_apps_v1_api()
+            if kind == "Deployment":
+                try:
+                    apps_v1.create_namespaced_deployment(
+                        namespace=namespace, body=manifest
+                    )
+                except kubernetes.client.rest.ApiException as e:
+                    if e.status == 409:  # Conflict
+                        print(f"Deployment {manifest_name} already exists. Patching...")
+                        apps_v1.patch_namespaced_deployment(
+                            name=manifest_name, namespace=namespace, body=manifest
+                        )
+                    else:
+                        raise
+        else:
+            custom_objects_api = self.get_custom_objects_api()
+            group, version = api_version.split("/")
+            plural = self._get_plural_name(kind)  # Helper needed for this
+
+            try:
+                custom_objects_api.create_namespaced_custom_object(
+                    group=group,
+                    version=version,
+                    namespace=namespace,
+                    plural=plural,
+                    body=manifest,
+                )
+            except kubernetes.client.rest.ApiException as e:
+                if e.status == 409:  # Conflict
+                    print(f"{kind} {manifest_name} already exists. Replacing...")
+                    # Need to get the resource version for replace
+                    try:
+                        existing = custom_objects_api.get_namespaced_custom_object(
+                            group=group,
+                            version=version,
+                            namespace=namespace,
+                            plural=plural,
+                            name=manifest_name,
+                        )
+                        manifest["metadata"]["resourceVersion"] = existing["metadata"][
+                            "resourceVersion"
+                        ]
+                        custom_objects_api.replace_namespaced_custom_object(
+                            group=group,
+                            version=version,
+                            namespace=namespace,
+                            plural=plural,
+                            name=manifest_name,
+                            body=manifest,
+                        )
+                    except kubernetes.client.rest.ApiException as get_e:
+                        print(f"Error getting existing object for replace: {get_e}")
+                        raise
+                else:
+                    raise
+
+    def _get_plural_name(self, kind: str) -> str:
+        # Basic pluralization, may need refinement
```

Comment:
Consider using the Kubernetes Discovery API to dynamically resolve the plural resource name for a given Kind, or use a library. You can add it as a TODO